src/js/pm-image-editor.js

```diff
@@ -0,0 +1,364 @@
+/*jshint multistr: true */
+(function () {
+    'use strict';
+
+    angular.module('pmImageEditor')
+        .factory('ImageEditorFactory', function () {
+            var ImageEditorFactory = function() {
+                // Visible area width. Image should always fit
+                // width or height (depends on rotation) to this area.
+                this.visibleWidth = 0;
+
+                // Set default ratio to make resetTransformations works correct.
+                this.ratio = 1;
+
+                // Real image width and height.
+                this.naturalWidth = 0;
+                this.naturalHeight = 0;
+
+                this.resetTransformations();
+            };
+
+            ImageEditorFactory.prototype.resetTransformations = function() {
+                // Coordinates of top-left corner.
+                this.top = 0;
+                this.left = 0;
+
+                // All the editor options.
+                this.selection = null;
+
+                // This variable contains rotation value by mod 2,
+                // for which latest crop was applyed.
+                this.wasCroppedForRotation = 0;
+
+                this.isCropped = false;
+                this.hFlip = false;
+                this.vFlip = false;
+
+                // Rotation value. Can be one of the following values:
+                //   0 - 0deg rotation,
+                //   1 - 90deg rotation,
+                //   2 - 180deg rotation,
+                //   3 - 270deg rotation.
+                this.rotation = 0;
```

Comment:
I'll try to change this. But it was easier to use 0-3 approach instead of degrees.